src/internal/clusterstate/v2.8.0/pfsdb.go

```diff
@@ -308,19 +327,155 @@ func createNotifyCommitsTrigger(ctx context.Context, tx *pachsql.Tx) error {
+		return errors.Wrap(err, "counting rows in collections.commits")
+	}
+	if count.Collections != count.Commits {
+		return errors.New(fmt.Sprintf("collections.commits has %d rows while pfs.commits has %d rows",
+			count.Collections, count.Commits))
+	}
+	pageSize := uint64(100)
+	totalPages := count.Collections / pageSize
+	if pageSize%count.Collections > 0 {
+		totalPages++
+	}
+	for i := uint64(0); i < totalPages; i++ {
+		page := make([]commitCollection, 0)
+		getCommitsQuery := `SELECT commit.int_id, col.key, col.proto, col.updatedat, col.createdat
+		FROM pfs.commits commit JOIN collections.commits AS col ON commit.commit_id = col.key 
+		ORDER BY commit.int_id ASC LIMIT %d OFFSET %d`
+		if err := env.Tx.SelectContext(ctx, &page, fmt.Sprintf(getCommitsQuery, pageSize, i*pageSize)); err != nil {
+			return errors.Wrap(err, "could not read table")
+		}
+		if len(page) == 0 {
+			return nil
+		}
+		if err := migratePage(ctx, env.Tx, page); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func migratePage(ctx context.Context, tx *pachsql.Tx, page []commitCollection) error {
+	migrateQuery := `WITH repo_row_id AS (SELECT id from pfs.repos WHERE name=:repo_name AND type=:repo_type 
+                                                AND project_id=(SELECT id from core.projects WHERE name= :proj_name))
+		UPDATE pfs.commits SET 
+			commit_id=:commit_id, 
+			commit_set_id=:commit_set_id,
+		    repo_id=(SELECT id from repo_row_id), 
+		    branch_id=(SELECT id from pfs.branches WHERE name=:branch_name AND repo_id=(SELECT id from repo_row_id)), 
+			description=:description, 
+			origin=:origin, 
+			start_time=:start_time, 
+			finishing_time=:finishing_time, 
+			finished_time=:finished_time, 
+		    compacting_time_s=:compacting_time_s, 
+			validating_time_s=:validating_time_s, 
+			size=:size, 
+			error=:error 
+		WHERE int_id=:int_id;`
+	for _, col := range page {
+		commit, err := protoToCommit(col)
+		if err != nil {
+			return err
+		}
+		_, err = tx.NamedExecContext(ctx, migrateQuery, commit)
+		if err != nil {
+			return errors.Wrap(err, fmt.Sprintf("migrating commit %d", commit.IntID))
+		}
+		if commit.ParentCommit == nil && commit.ChildCommits == nil {
+			continue
+		}
+		if err := migrateRelatives(ctx, tx, commit); err != nil {
+			return errors.Wrap(err, fmt.Sprintf("migrating relatives of commit %d", commit.IntID))
+		}
+	}
+	return nil
+}
+
+func protoToCommit(col commitCollection) (*Commit, error) {
+	commitInfo := &pfs.CommitInfo{}
+	if err := proto.Unmarshal(col.Proto, commitInfo); err != nil {
+		return nil, errors.Wrapf(err, "could not unmarshal proto")
+	}
+	if commitInfo.Details == nil {
+		commitInfo.Details = &pfs.CommitInfo_Details{}
+	}
+	if commitInfo.Commit.Branch == nil {
+		commitInfo.Commit.Branch = &pfs.Branch{Name: "master", Repo: commitInfo.Commit.Repo}
+	}
+	return &Commit{
+		IntID:          col.IntID,
+		CommitID:       pfsdb.CommitKey(commitInfo.Commit),
```

Comment:
To avoid explicit dependency on pfsdb, you can use the recently added:
```suggestion
		CommitID:       commitInfo.Commit.Key(),
```